bindings/cs/rl.net.cli/RLSimulator.cs

```diff
@@ -322,5 +322,29 @@ public void Run(int steps = SimulatorStepProvider.InfinitySteps, int slots = Sim
                 this.driver.OnError -= value;
             }
         }
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
```

Comment:
just for my education: is this some kind of C# object disposal trick? why not put stuff in `Dispose()`?

Reply:
It's not a trick, it's the recommended way to implement IDisposable.

It avoids a lot of common pitfalls associated with implementing it.

Here's an article explaining it:
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose